source/mainnet/smart-contracts/tutorials/wCCD/wCCD-front-end-set-up.rst

```diff
@@ -4,6 +4,354 @@
 Setting up the front-end
 ========================
 
-Coming soon.
+Non-tech users might find it inconvenient to interact via the Concordium node with smart contracts
+and some users may choose not to host their own Concordium node in the future.
+
+You can lower the bar for entry by coding an appealing front-end that provides additional information
+to your potential customers. Web front-ends are a familiar sight nowadays, but keep in mind
+that downloading a crypto wallet as a browser extension and its behavior or safe usage might be new
+for people using your front-end. Providing comprehensive explanations and step-by-step guides on your website on topics
+related to the browser wallet is important for a good user experience. The browser wallet
+connects via ``HTTPS`` to a server that is connected to a Concordium node. This setup alleviates the
+need for the user to host their own Concordium node.
+
+.. note::
+
+    Before you start with part 3 of this tutorial, make sure you have:
+
+    - access to the `Chrome <https://www.google.com/chrome/>`_ web browser on your laptop or computer
+
+    - `git <https://git-scm.com/>`_ installed to be able to clone a repository from `GitHub <https://github.com/>`_
+
+    - `node.js <https://nodejs.org/en/download/>`_ and `yarn <https://yarnpkg.com/getting-started/install>`_ installed
+
+    ``Node.js`` is a JavaScript runtime environment and ``yarn`` is a package manager
+    used in combination with ``node.js``. You can download `node.js <https://nodejs.org/en/download/>`_  and
+    `yarn <https://yarnpkg.com/getting-started/install>`_  on their official websites.
+
+Concordium browser wallet
+-------------------------
+
+A browser wallet is a piece of code that can be added as an extension to supported browsers such as ``Chrome``.
+The browser wallet hosts the private keys corresponding to the accounts of the user and a link that points
+to a `JSON-RPC server  <https://github.com/Concordium/concordium-json-rpc>`_.
+
+Your front-end code that is run in the browser constructs the transaction object
+and sends it the the browser wallet. The transaction object is signed by the private key hosted in the browser wallet
+and transmitted to the ``JSON-RPC server`` via ``HTTPS``. This server has access to a Concordium node and masks
```

Comment:
What does "masks" mean here? I don't understand it.

Reply:
It means to change the HTTPS request format into a JSON-RPC request format that the node understands.

Reply:
I use `converts` now which seems clearer than `masks`